### DISCUSSION TITLE
Experience from writing modules

### DISCUSSION THREAD

**Discussion by elsalant** [fybrik/fybrik#1159] [Ideas]
I have now written a number of Fybrik modules.   Here are some of my impressions with the aim of seeing how we can simplify this process.

[Observations writing modules.docx](https://github.com/fybrik/fybrik/files/7903629/Observations.writing.modules.docx)

**Comment by shlomitk1** [comment]
Some of the issues are around Helm charts. This link https://helm.sh/docs/chart_template_guide/getting_started/ explains the basic chart structure and helm concepts. This site can be reached from the main Fybrik site: https://fybrik.io/v0.6/contribute/modules/#module-helm-chart.
Credential management: https://fybrik.io/v0.6/contribute/modules/#credential-management
https://github.com/fybrik/fybrik/issues/1168 resolved by Sima provides a better explanation for the manager being unable to construct the data path
Connection properties (e.g. bucket, endpoint) are defined by the taxonomy.
